deeppavlov/core/models/torch_model.py

```diff
@@ -157,16 +161,29 @@ def load(self, fname: Optional[str] = None, *args, **kwargs) -> None:
                 log.info(f"Initializing `{self.__class__.__name__}` from saved.")
 
                 # firstly, initialize with random weights and previously saved parameters
-                self.init_from_opt(model_func)
+                if model_func is not None:
+                    self.init_from_opt(model_func)
 
                 # now load the weights, optimizer from saved
                 log.info(f"Loading weights from {weights_path}.")
                 checkpoint = torch.load(weights_path, map_location=self.device)
+                model_state = checkpoint["model_state_dict"]
+                optimizer_state = checkpoint["optimizer_state_dict"]
+
+                # load a multi-gpu model on a single device
+                if not self.is_data_parallel and "module." in list(model_state.keys())[0]:
+                    tmp_model_state = {}
+                    for key, value in model_state.items():
+                        tmp_model_state[re.sub("module.", "", key)] = value
+                    model_state = tmp_model_state
+
+                strict_load_flag = bool([key for key in checkpoint["model_state_dict"].keys()
+                                         if key.endswith("embeddings.position_ids")])
```

Comment:
strict_load_flag = any([key.endswith(...) for key in checkpoint['model_state_dict']])